File: woodroid/src/main/java/me/gilo/woodroid/data/api/CouponAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.Product;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public interface CouponAPI{

    @Headers("Content-Type: application/json")
    @POST("coupons")
    Call<Coupon> create(@Body Coupon body);

    @GET("coupons/{id}")
    Call<Coupon> view(@Path("id") int id);

    @GET("coupons")
    Call<List<Coupon>> list();

    @GET("coupons")
    Call<ArrayList<Coupon>> filter(@QueryMap Map<String, String> filter);

    @Headers("Content-Type: application/json")
    @PUT("coupons/{id}")
    Call<Coupon> update(@Path("id") int id, @Body Coupon body);

    @DELETE("coupons/{id}")
    Call<Coupon> delete(@Path("id") int id);

    @DELETE("coupons/{id}")
    Call<Coupon> delete(@Path("id") int id, @Query("force") boolean force);

    @POST("coupons/batch")
    Call<String> batch(@Body Coupon body);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/CouponAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.Coupon
import retrofit2.Call
import retrofit2.http.*

interface CouponAPI {

    @Headers("Content-Type: application/json")
    @POST("coupons")
    fun create(@Body body: Coupon): Call<Coupon>

    @GET("coupons/{id}")
    fun view(@Path("id") id: Int): Call<Coupon>

    @GET("coupons")
    fun list(): Call<List<Coupon>>

    @GET("coupons")
    fun filter(@QueryMap filter: Map<String, String>): Call<List<Coupon>>

    @Headers("Content-Type: application/json")
    @PUT("coupons/{id}")
    fun update(@Path("id") id: Int, @Body body: Coupon): Call<Coupon>

    @DELETE("coupons/{id}")
    fun delete(@Path("id") id: Int): Call<Coupon>

    @DELETE("coupons/{id}")
    fun delete(@Path("id") id: Int, @Query("force") force: Boolean): Call<Coupon>

    @POST("coupons/batch")
    fun batch(@Body body: Coupon): Call<String>

}